src/geometry/mod.rs

```diff
@@ -107,8 +107,8 @@ impl ShapeCastHit {
         hit: rapier::parry::query::ShapeCastHit,
         details_always_computed: bool,
     ) -> Self {
-        let details = if !details_always_computed
-            && hit.status != ShapeCastStatus::PenetratingOrWithinTargetDist
+        let details = if details_always_computed
+            || hit.status != ShapeCastStatus::PenetratingOrWithinTargetDist
```

Comment:
After consideration, I think I'd prefer :

```rs
        let details = match (details_always_computed, hit.status) {
            (_, ShapeCastStatus::Failed) => None,
            (false, ShapeCastStatus::PenetratingOrWithinTargetDist) => None,
            _ => Some(ShapeCastHitDetails {
                witness1: hit.witness1.into(),
                witness2: hit.witness2.into(),
                normal1: hit.normal1.into(),
                normal2: hit.normal2.into(),
            }),
        };
```

Reply:
or is alternative condition:
```rs
hit.status == ShapeCastStatus::OutOfIterations) ||
hit.status == ShapeCastStatus::Converged) ||  
(details_always_computed && hit.status == ShapeCastStatus::PenetratingOrWithinTargetDist)

Reply:
Changed code to exactly as you written in the first comment